From 80ddbec24c3088ccdb1345ff2039022ad3d7100d Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Thu, 16 Nov 2023 17:08:10 +0100
Subject: [PATCH 0110/1013] net: dovetail: pass net struct to oob_state
 init/cleanup hooks

The companion core may want to refer to any field within the net
namespace to initialize/cleanup for oob support.  Change the signature
of the net_{init, cleanup}_oob_state() hooks accordingly.

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 include/dovetail/net.h   | 6 ++++--
 net/core/net_namespace.c | 4 ++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/include/dovetail/net.h b/include/dovetail/net.h
index 163fbccd862d..c7ef00fb9a5f 100644
--- a/include/dovetail/net.h
+++ b/include/dovetail/net.h
@@ -9,7 +9,9 @@
 struct oob_net_state {
 };
 
-static inline void net_init_oob_state(struct oob_net_state *nets) { }
-static inline void net_cleanup_oob_state(struct oob_net_state *nets) { }
+struct net;
+
+static inline void net_init_oob_state(struct net *net) { }
+static inline void net_cleanup_oob_state(struct net *net) { }
 
 #endif /* !_DOVETAIL_NET_H */
diff --git a/net/core/net_namespace.c b/net/core/net_namespace.c
index b3a20e847140..bb92d16245a0 100644
--- a/net/core/net_namespace.c
+++ b/net/core/net_namespace.c
@@ -352,7 +352,7 @@ static __net_init int setup_net(struct net *net)
 	net->net_cookie = gen_cookie_next(&net_cookie);
 	preempt_enable();
 	/* Dovetail: set up the oob-specific state. */
-	net_init_oob_state(&net->oob);
+	net_init_oob_state(net);
 
 	list_for_each_entry(ops, &pernet_list, list) {
 		error = ops_init(ops, net);
@@ -618,7 +618,7 @@ static void cleanup_net(struct work_struct *work)
 		unhash_nsid(net, last);
 		list_add_tail(&net->exit_list, &net_exit_list);
 		/* Dovetail: drop the oob-specific state. */
-		net_cleanup_oob_state(&net->oob);
+		net_cleanup_oob_state(net);
 	}
 
 	/* Run all of the network namespace pre_exit methods */
-- 
2.30.2

